### PR TITLE
feat: Initial driver setup

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This is another try to finish https://github.com/appium/appium-geckodriver/pull/1

cc @vrunoa

### CODE REVIEW

**Review by jlipps — APPROVED** [review]
There's now a decent amount of duplication between this and safari driver. In the future would it make sense to have a basic proxy driver base class? Seems like all the web drivers would overlap in logic in this way.

Also, given that Geckodriver is not included on the platform like safaridriver is, as a future feature we could implement downloading of geckodriver like we do for chromedriver, so that it works out of the box for appium users.

**Comment by vrunoa** [comment]
Awesome @mykola-mokhnach

**Review by vrunoa — APPROVED** [review]

**Comment by mykola-mokhnach** [comment]
For now I just wanted to make it as simple as possible. The logic is quite different from Safari driver, because there we keep single instance per Appium process and here we spin a new instance for each session

File: lib/gecko.js [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        await this.proc.stop('SIGKILL');
+      } catch (ign) {}
+    }
+  }
+}
+
+const RUNNING_PROCESS_IDS = [];
+process.once('exit', () => {
+  for (const pid of RUNNING_PROCESS_IDS) {
+    const command = system.isWindows() ? `taskkill.exe /PID ${pid} /F` : `kill ${pid}`;
```
Comment (KazuCocoa) [inline lib/gecko.js:117]: `/F` should be a fallback method on Windows. The `/F` sometimes shows an error dialog which requires GUI interactions on Windows. (Not always on my experience)
So, try without `/F` -> but the process still remains after a few seconds -> then calls `/F` is pretty better on Windows.

**Review by KazuCocoa — COMMENTED** [review]
btw, i enabled Azure https://dev.azure.com/AppiumCI/Appium%20CI/_build?definitionId=49

File: lib/gecko.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/gecko.js:117]: I'll remove it for now completely. will add later if we have any issues there

**Pull request merged**